Segments/Blackburrow.mapproj

```diff
@@ -796,6 +796,92 @@
         
     }
     
+    public partial class RJavelin: Staff, IReturningWeapon, ITreasure
+    {
+        /// <inheritdoc />
+        public override int LabelNumber => 6000053;
+
+        /// <inheritdoc />
+        public override uint BasePrice => 10000;
+
+        /// <inheritdoc />
+        public override int Weight => 2560;
+
+        /// <inheritdoc />
+        public override int MinimumDamage => 3;
+
+        /// <inherit />
+        public override Skill Skill => Skill.Staff;
+
+        /// <inheritdoc />
+        public override int MaximumDamage => 7;
+
+        /// <inheritdoc />
+        public override int BaseArmorBonus => 2; 
+
+        /// <inheritdoc />
+        public override int BaseAttackBonus => 5;
+        
+        /// <inheritdoc />
+        public override bool CanDisintegrate => false;
+
+        /// <inheritdoc />
+        public override WeaponFlags Flags => WeaponFlags.BlueGlowing | WeaponFlags.Throwable | WeaponFlags.Piercing | WeaponFlags.Returning | WeaponFlags.Lawful;
+
+        /// <inheritdoc />
+        public override bool CanBind => true;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="RJavelin"/> class.
+        /// </summary>
+        public RJavelin() : base(956)
+        {
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="RJavelin"/> class.
+        /// </summary>
+        public RJavelin(Serial serial) : base(serial)
+        {
+        }
+
+        /// <inheritdoc />
+        public override void GetDescription(List<LocalizationEntry> entries)
+        {
+            entries.Add(new LocalizationEntry(6200000, 6200157)); 
+        }
+
+        /// <inheritdoc />
+        public override void Serialize(BinaryWriter writer)
+        {
+            base.Serialize(writer);
+
+            writer.Write((short)1); /* version */
+        }
+
+        /// <inheritdoc />
+        public override void Deserialize(BinaryReader reader)
+        {
+            base.Deserialize(reader);
+
+            var version = reader.ReadInt16();
+
+            switch (version)
+            {
+                case 1:
+                    {
+                        break;
+                    }
+            }
+        }
+        
+        public override void OnHit(MobileEntity attacker, MobileEntity defender)
+        {
+            attacker.CalculateThrowDamage *= (source.GetDistanceToMax(entity) + 1);
```

Comment:
attacker.CalculateThrowDamage *= (attacker.GetDistanceToMax(defender) + 1);

Reply:
Ok, so I need to specify that it's defender there?